andela/vidar-ah-backend
# #164693379 Create, update and delete category by admin and super admin

#### What does this PR do?
This PR implements user roles (user, admin and superAdmin) and restricts the privilege of creating, updating, and deleting categories to only admin and super admin.

#### Description of Task to be completed?
Admin and SuperAdmin should be able to create, update or delete a category for articles.

#### How should this be manually tested?
- Check .env.sample for the environment variables needed to test the application
- Run `npm run db-migrate` to run migrations
- Run `npm run build` to build the application
- Run `npm start` to start the application
- Sign in with super admin email and password
- To create a category, make a POST request to `/api/v1/category` with a JSON object containing the field: categoryName. If the `categoryName` is not provided, a 422 error message with be returned. If this request is successful, a response with status 201, a success message, and the newly created category will be returned. 
- To update a category, make a patch request to `/api/v1/category/<categoryID>` with a JSON object containing the new value for categoryName.  If the `categoryName` is not provided, a 422 error message with be returned. If this request is successful, a response with status 200 and a success message will be returned.
- To delete a category, hit the endpoint `/api/v1/category/<categoryID>` with the DELETE HTTP method. 

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164693379

#### Screenshots (if appropriate)
- N/A

#### Questions:
N/A

[Finishes #164693379]
